projects/plugins/boost/app/modules/cache/Boost_Cache_Settings.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+/*
+ * Cache settings class.
+ * Settings are stored in a file in the boost-cache directory.
+ * This file is shared by autoloaded code and code loaded by advanced-cache.php
+ */
+
+class Boost_Cache_Settings {
+	private $settings   = array();
+	private $last_error = '';
+	private $config_file_path;
+	private $config_file;
+
+	public function __construct() {
+		$this->config_file_path = WP_CONTENT_DIR . '/boost-cache/';
+		$this->config_file      = $this->config_file_path . 'config.php';
+		$this->init_settings();
+	}
+
+	/*
+	 * Load the settings from the config file, if available.
+	 */
+	private function init_settings() {
+
+		if ( ! file_exists( $this->config_file_path ) ) {
+			mkdir( $this->config_file_path, 0755, true ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_mkdir
+		}
+
+		if ( ! file_exists( $this->config_file ) ) {
+			$result = $this->set( array( 'enabled' => true ) );
+			// TODO: handle $result if it's false
```

Comment:
Is this a left over or is it something that needs doing?

Reply:
Oh yeah, I meant to take that out.

Reply:
I left it in, but slightly modified it and made "enabled" = false. We don't check that "enabeld" settings yet so it does nothing right now.